### ISSUE TITLE
apache::mod_php on ubuntu 16.04 creates misnamed config files

### ISSUE THREAD

**Issue by carangil** [sous-chefs/apache2#480]
### Cookbook version

Berks says "Using apache2 (3.3.0)"

### Chef-client version

12.18.31

### Platform Details

Ubuntu 16.04 in AWS

### Scenario:

When using mod_php, Ubuntu creates:
  /etc/apache2/mods-available/php7.0.load and /etc/apache2/mods-available/php7.0.conf
The recipe overwrites them with stub files w/ comment saying to use different config file instead.  It them creates:
  /etc/apache2/mods-available/php7.load and unmatching  /etc/apache2/mods-available/php.conf (no seven)
Then a2enmod creates the mods-enabled symlink for LOAD but not for CONF, so the module is loaded, but not configured to run for .php files.  As a result, navigating a php document in a browser downloads it instead of running it.

After the recipe is run, renaming php.conf manually to php7.conf and restarting apache fixes the issue.

If I don't run chef, and instead install apache2 and libapache2-mod-php using Ubuntu apt-get, this issue does not occur.

### Steps to Reproduce:

Create recipe including apache2 and apache2::mod_php 
Run recipe on Ubuntu 16.04
To to navigate a .php file placed in /var/www/html; it will be displayed in the browser instead of executed

### Expected Result:
php file executed on server, output displayed to web browser

### Actual Result:
php file source code displayed in browser

**Comment by svanzoest** [comment]
Thanks for the detailed report. I think this is a duplicate of #479

**Comment by jamescorkran** [comment]
Has this been fixed in the current build since there are now multiple issues with it?

**Comment by fancyweb** [comment]
Having the same problem on Debian Stretch (9.0)

This is my fix : `ln -s /etc/apache2/mods-available/php.conf /etc/apache2/mods-enabled/php7.conf`

**Comment by jamescorkran** [comment]
@fancyweb 
I solved mine by flipping [this recipe setting to true](https://github.com/sous-chefs/apache2/blob/master/recipes/mod_php.rb#L99)
Then duplicated [apache/templates/default/mods/php.conf.erb](https://github.com/sous-chefs/apache2/blob/master/templates/default/mods/php.conf.erb), tweaked it and renamed it to php7.conf.erb and that seems to have worked for me.

**Comment by jarrettj** [comment]
Hi,

Has the installation of libapache2-mod-php on Ubuntu Xenial been sorted?

Regards.
JJ

**Comment by eyespies** [comment]
This is also broken on CentOS 7:

```
[vagrant@magento httpd]$ ls mods-available/php*
mods-available/php5.load  mods-available/php.conf
[vagrant@magento httpd]$ ls mods-enabled/php*
mods-enabled/php5.load
```

So PHP does not work when the system comes online. For now, I'm just creating a symlink in my wrapper recipe, but would be good to get this fixed.

**Comment by loforbes** [comment]
Just want to echo this is broken on CentOS 7, specifically when trying to use the php5 module. The mods-enabled/php5.load link is created. However, no link is created for mods-enabled/php5.conf, nor is there a mods-available/php5.conf to link to. We manually linked mods-enabled/php5.conf -> mods-available/php.conf and the service launches properly.
Regards,
-liam

**Comment by fabn** [comment]
This is a big issue and after 2 months is still open. Could we fix it in the cookbook?

Thanks.

**Comment by josephholsten** [comment]
@fabn absolutely, do you have a PR ready for me to review? I don’t use this cookbook on a daily basis, but I’m happy to help any changes you make become ready to merge and release.

**Comment by pwalz** [comment]
It looks like this is fixed by #514, which has been merged.
